haystack/dataclasses/chat_message.py

```diff
@@ -99,3 +99,54 @@ def from_function(cls, content: str, name: str) -> "ChatMessage":
         :returns: A new ChatMessage instance.
         """
         return cls(content, ChatRole.FUNCTION, name)
+
+    def to_dict(self, flatten: bool = True) -> Dict[str, Any]:
```

Comment:
Why do we need a `flatten` parameter at all?

Reply:
I added it to keep consistency, as it is provided by the 'Document' serialization functionality.

Reply:
Let's remove it for now as it adds more complexity without serving an immediate need.

Reply:
I removed the parameter, but still support flattened and non-flattened metadata at deserialization.